pkg/cmd/roachtest/tests/kv.go

```diff
@@ -79,6 +79,24 @@ func registerKV(r registry.Registry) {
 			return opts.splits
 		}
 	}
+	computeConcurrency := func(opts kvOptions) int {
+		// Scale the workload concurrency with the number of nodes in the cluster to
+		// account for system capacity, then scale inversely with the batch size to
+		// account for the cost of each operation.
+		// TODO(nvanbenschoten): should we also scale the workload concurrency with
+		// the number of CPUs on each node? Probably, but doing so will disrupt
+		// regression tracking.
+		concPerNode := 64
+		if opts.concMultiplier != 0 {
+			concPerNode = opts.concMultiplier
+		}
+		conc := concPerNode * opts.nodes
+		const batchOpsPerConc = 2
```

Comment:
How did you settle on the 2 in conc /= (batchSize/2)?

Reply:
I found that if I just divided by concurrency, we pulled too far away from overload and the system was no longer saturated. This value struck a good balance.